include/fmt/core.h

```diff
@@ -845,11 +845,23 @@ template <typename T = char> class counting_buffer : public buffer<T> {
 // It is used to reduce symbol sizes for the common case.
 template <typename T>
 class buffer_appender : public std::back_insert_iterator<buffer<T>> {
+  using base = std::back_insert_iterator<buffer<T>>;
  public:
   explicit buffer_appender(buffer<T>& buf)
-      : std::back_insert_iterator<buffer<T>>(buf) {}
-  buffer_appender(std::back_insert_iterator<buffer<T>> it)
-      : std::back_insert_iterator<buffer<T>>(it) {}
+      : base(buf) {}
+  buffer_appender(base it)
+      : base(it) {}
+
+  buffer_appender& operator++() {
+    base::operator++();
+    return *this;
+  }
+
+  buffer_appender operator++(int) {
+    buffer_appender tmp = *this;
+    ++*this;
+    return tmp;
+  }
```

Comment:
Given that `back_insert_iterator` only has like... 4 member functions, and I've already implemented 2 of them (and over-implemented them too, these could both just be `return *this;`), maybe we should just implement the whole thing in here without `std`?

Reply:
I think it's fine as is.